enhancements/installer/signal-cluster-deletion.md

```diff
@@ -0,0 +1,152 @@
+---
+title: signal-cluster-deletion
+authors:
+  - "@abutcher"
+reviewers:
+  - dgoodwin
+  - abhinavdahiya
+  - wking
+  - eparis
+  - ecordell
+
+approvers:
+  - dgoodwin
+  - abhinavdahiya
+  - wking
+  - eparis
+creation-date: 2020-03-30
+last-updated: 2020-05-07
+status: provisional
+---
+
+# Signal Cluster Deletion
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Operators can create external resources in clouds and other services that have no opportunity to be cleaned up when a cluster is uninstalled, causing extra work for cluster maintainers to track down and remove these resources. Signalling cluster deletion would allow operators to clean up external resources during uninstallation and would ensure that total cleanup is an automatic process.
```

Comment:
This is probably inherently a best-effort process, and there is a risk that people might interpret it as a guaranteed process.  An in-cluster mechanism cannot survive people getting fed up waiting for clean-up to happen, taking things into their own hands and hard deleting the cluster themselves.  (Think when you last kill -9'ed a process that was taking too long to quit).